src/raw_stream.rs

```diff
@@ -493,6 +520,96 @@ impl RawDevice {
             .map_err(nix_err)
     }
 
+    /// Update the auto repeat delays
+    #[inline]
+    pub fn update_auto_repeat(&self, repeat: &AutoRepeat) -> io::Result<()> {
+        let repeat = [repeat.delay, repeat.period];
+
+        unsafe { sys::eviocsrep(self.as_raw_fd(), &repeat) }
+            .map(|_| ())
+            .map_err(nix_err)
+    }
+
+    /// Retrieve the scancode for a keycode, if any
+    pub fn get_scancode_by_keycode(&self, keycode: u32) -> io::Result<Vec<u8>> {
+        let mut keymap = libc::input_keymap_entry {
+            flags: 0,
+            len: 0,
+            index: 0,
+            keycode,
+            scancode: [0u8; 32],
+        };
+
+        unsafe { sys::eviocgkeycode_v2(self.as_raw_fd(), &mut keymap) }
+            .map(|_| keymap.scancode[..keymap.len as usize].to_vec())
+            .map_err(nix_err)
+    }
+
+    /// Retrieve the keycode and scancode by index, starting at 0
+    pub fn get_scancode_by_index(&self, index: u16) -> io::Result<(u32, Vec<u8>)> {
+        let mut keymap = libc::input_keymap_entry {
+            flags: 1,
```

Comment:
Does this 1 correspond to a constant somewhere? `INPUT_KEYMAP_BY_INDEX` it looks like?

Reply:
Yes, that is exactly it. The libc crate has no define for it. I'll add a constant.